vignettes/simplify_function.Rmd

```diff
@@ -1,28 +1,121 @@
 
-When you click the **Knit** button a document will be generated that includes both content as well as the output of any embedded R code chunks within the document. You can embed an R code chunk like this:
+## Simplify Prediction Results
 
-```{r cars}
-summary(cars)
+First of all, let's define our feature columns, input function, and a basic linear classifier:
+
+```{r}
+library(tfestimators)
+
+cols <- feature_columns( 
+  column_numeric("disp", "cyl")
+)
+
+mtcars_input_fn <- function(data) {
+  input_fn(data, 
+           features = c("disp", "cyl"), 
+           response = "vs")
+}
+
+indices <- sample(1:nrow(mtcars), size = 0.80 * nrow(mtcars))
+train_data <- mtcars[indices, ]
+test_data  <- mtcars[-indices, ]
+
+model <- linear_classifier(feature_columns = cols)
+```
+
+and we train the model using the input function constructed from training data:
+
+```{r}
+model %>% train(mtcars_input_fn(train_data))
+```
+
+Here's how the prediction output looks like without applying the simplifier on the results:
+
+```{r}
+model %>% predict(mtcars_input_fn(mtcars[1, ]), simplify = FALSE)
+```
+
+Note that here we only evaluate the first row of `mtcars` data since the results will be in forms of long nested lists that may not fit in this page.
+
+
+### Default Simplifier
+
+If `simplify = TRUE`, a default simplify function for predictions will be used to flatten the results so they are more appearing and concise like the following:
+
+```{r}
+model %>% predict(mtcars_input_fn(test_data), simplify = TRUE)
+```
+
+Note that this time we use the full test dataset since no matter how large the prediction results are, we can still see the results in a concise [tibble object](http://tibble.tidyverse.org/index.html). 
+
+### Custom Simplifier for Prediction Result
+
+You can also write your own custom simplify function to clean up the prediction results. For example, below we define a custom function that obtains the probabilities from the list of prediction results with keys being the prediction keys pre-defined for this canned estimator. In this case, `linear_classifier` has logits, probabilities, prediction classes, logistics, etc. pre-defined. We can see a full list of available predictions keys by calling `prediction_keys()`. 
+
+```{r}
+simplify_fn <- function(predictions) {
+  lapply(predictions, function(x) x$probabilities)
+}
+
+model %>% predict(mtcars_input_fn(mtcars[1:2, ]), simplify = simplify_fn)
+```
+Note again that we only make predictions for the first two rows of the data since it's too verbose to fit in this page.
+
+## Simplify Evaluation Results
+
+Similarly, the evaluation results can be simplified as well. Here's how the raw evaluation results look like without simplification by providing `simplify = FALSE`. 
```

Comment:
"Evaluation results can also be simplified. Here's what the raw evaluation results look like without simplification.."